com.unity.render-pipelines.core/Runtime/Textures/MSAASamples.cs

```diff
@@ -14,4 +18,52 @@ public enum MSAASamples
         /// <summary>MSAA 8X.</summary>
         MSAA8x = 8
     }
+
+    public static class MSAASamplesExtensions
+    {
+        public static FormatUsage AsFormatUsage(this MSAASamples samples)
+        {
+            switch (samples)
+            {
+                case MSAASamples.MSAA2x:
+                    return FormatUsage.MSAA2x;
+                case MSAASamples.MSAA4x:
+                    return FormatUsage.MSAA4x;
+                case MSAASamples.MSAA8x:
+                    return FormatUsage.MSAA8x;
+                default:
+                    return FormatUsage.Render;
+            }
+        }
+
+        public static MSAASamples Validate(this MSAASamples samples)
+        {
+            if (samples == MSAASamples.None)
+                return samples;
+
+            GraphicsFormat ldrFormat = SystemInfo.GetGraphicsFormat(DefaultFormat.LDR);
+            MSAASamples[] probe = {MSAASamples.MSAA2x, MSAASamples.MSAA4x, MSAASamples.MSAA8x};
```

Comment:
This is dangerous, if we add MSAA16x in the future this will not be catch, we should generate the array from the enum .
i.e var valuesAsArray = Enum.GetValues(typeof(Enumnum));

Reply:
How are you suggesting to automatically handle MSAASamples.MSAAx mapping into FormatUsage.MSAAx that would still be a problem?